### PR TITLE
increased regex query speed

### PR DESCRIPTION
**Pull request by zwzn**

### CODE REVIEW

File: comicboxd/app/database/funcs.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
-var debug = false
+var debug = true
```
Comment (buckley-w-david) [inline comicboxd/app/database/funcs.go]: 🔥

**Comment by zwzn** [comment]
I need to stop doing that

**Comment by buckley-w-david** [comment]
This is kind of a dense bit of changes, correct me if I'm wrong here.

The idea behind this change is some extra logic that figures out if you can skip doing an actual regex check and can instead get away with doing a `LIKE` query?

**Comment by zwzn** [comment]
That's exactly what it does. The like queries are much faster than the regex ones. Form my full book db it shrinks the series query from 2s to 120ms

**Comment by buckley-w-david** [comment]
Don't know if it's somehow due to these changes (I actually doubt it is), but my console is accumulating these:
```
2018/12/31 13:51:44 http: multiple response.WriteHeader calls
E: 13:51:44 write tcp [::1]:8081->[::1]:44532: write: broken pipe

goroutine 24 [running]:
runtime/debug.Stack(0xc00055a0e0, 0x1f4, 0x8000000080)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0xa7
github.com/zwzn/comicbox/comicboxd/app/middleware.writeError(0xc18f60, 0xc00055a0e0, 0xad3640, 0xc00054b130, 0x1f4)
	/home/david/Projects/comicbox/comicboxd/app/middleware/response.go:55 +0x5b
github.com/zwzn/comicbox/comicboxd/app/middleware.Recover.func1.1(0xc18f60, 0xc00055a0e0)
	/home/david/Projects/comicbox/comicboxd/app/middleware/recover.go:18 +0x6f
panic(0xad3640, 0xc00054b130)
	/usr/lib/go/src/runtime/panic.go:513 +0x1b9
github.com/zwzn/comicbox/comicboxd/errors.Check(...)
	/home/david/Projects/comicbox/comicboxd/errors/httpError.go:89
github.com/zwzn/comicbox/comicboxd/app/controller.(*book).Page(0x108b960, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/controller/bookController.go:100 +0xaa5
github.com/zwzn/comicbox/comicboxd/app/controller.(*book).Page-fm(0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/routes/web.go:31 +0x48
net/http.HandlerFunc.ServeHTTP(0xc0001f8200, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/usr/lib/go/src/net/http/server.go:1964 +0x44
github.com/zwzn/comicbox/comicboxd/app/middleware.HTTPCache.func1(0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/middleware/cache.go:78 +0xb1
net/http.HandlerFunc.ServeHTTP(0xc000249840, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/usr/lib/go/src/net/http/server.go:1964 +0x44
github.com/zwzn/comicbox/comicboxd/app/middleware.Auth.func1(0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/middleware/auth.go:52 +0x152
net/http.HandlerFunc.ServeHTTP(0xc000249860, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/usr/lib/go/src/net/http/server.go:1964 +0x44
github.com/zwzn/comicbox/comicboxd/app/middleware.Session.func1(0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/middleware/session.go:26 +0x11d
net/http.HandlerFunc.ServeHTTP(0xc000249880, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/usr/lib/go/src/net/http/server.go:1964 +0x44
github.com/zwzn/comicbox/comicboxd/app/middleware.Response.func1(0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/middleware/response.go:19 +0xa0
net/http.HandlerFunc.ServeHTTP(0xc0002498a0, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/usr/lib/go/src/net/http/server.go:1964 +0x44
github.com/zwzn/comicbox/comicboxd/app/middleware.Log.func1(0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/middleware/log.go:15 +0x90
net/http.HandlerFunc.ServeHTTP(0xc0002498c0, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/usr/lib/go/src/net/http/server.go:1964 +0x44
github.com/zwzn/comicbox/comicboxd/app/middleware.Recover.func1(0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/Projects/comicbox/comicboxd/app/middleware/recover.go:21 +0x95
net/http.HandlerFunc.ServeHTTP(0xc0002498e0, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/usr/lib/go/src/net/http/server.go:1964 +0x44
github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001fc000, 0xc18f60, 0xc00055a0e0, 0xc00043ad00)
	/home/david/go/pkg/mod/github.com/gorilla/mux@v1.6.2/mux.go:162 +0xf1
net/http.serverHandler.ServeHTTP(0xc000202000, 0xc18f60, 0xc00055a0e0, 0xc00043a500)
	/usr/lib/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc0002132c0, 0xc19620, 0xc000207c80)
	/usr/lib/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve
	/usr/lib/go/src/net/http/server.go:2851 +0x2f5
```

You ever see that/know what it's about?

I'm not noticing anything wrong actually using it, but my terminal being full of red is generally a bad sign.

**Comment by zwzn** [comment]
It looks like the network connection gets closed part way through encoding the image. I can't replicate this, where is it happening?

**Comment by zwzn** [comment]
I think that it is the images getting canceled partway through loading them. I don't think this is an issue from this PR. I am also getting it on master

**Pull request merged**